chore(Cross): [IOAPPX-322] Update the entire `Assistance` section with the new DS components

dmnplb opened the pull request:
> [!warning]
> This PR depends on:
> * https://github.com/pagopa/io-app-design-system/pull/290
> * https://github.com/pagopa/io-app-design-system/pull/291
> * https://github.com/pagopa/io-app-design-system/pull/294

## Short description
This PR updates the entire `Assistance` section with the new DS components.

## List of changes proposed in this pull request
- Remove `BaseScreenComponent` from existent screens
- Handle the different headers with `react-navigation`
- Add the new components, like `FooterActions` or `IOScrollViewWithLargeHeader` to replace the legacy components
- Use `FlatList` with appropriate `ListItem…` or `AccordionItem` components instead of `.map()` and custom components
- Update `IOScrollViewWithLargeHeader` to include a section reference above the title
- Use `OperationResultScreenContent` to render the empty state when there are no open requests
- Remove legacy `IdpCustomContextualHelpContent` component

### Open issues
-  `PanicMode` and other error screens still do not have updates

### Preview
| First page | Choose category | Ask permissions |
|--------|--------|--------|
| <video src="https://github.com/pagopa/io-app/assets/1255491/07d113be-262d-4245-84e0-a72b143a372d" /> | <video src="https://github.com/pagopa/io-app/assets/1255491/771a2b69-9413-4d8d-b4ce-8d6f84c06dda" /> | <video src="https://github.com/pagopa/io-app/assets/1255491/bf74fc8e-3ac9-4cd4-b729-0056755c93c2" /> | 


## How to test
Test different **Assistance** flows from different sections

shadowsheep1 commented on ts/components/screens/BaseScreenComponent/utils.tsx:
```diff
@@ -40,24 +38,16 @@ export const handleOnLinkClicked = (hideHelp: () => void) => (url: string) => {
-  onLinkClicked: (url: string) => void,
-  shouldHandleLink?: (url: string) => boolean
+  contextualHelpMarkdown: ContextualHelpPropsMarkdown | undefined
+  // onLoadEnd: () => void,
+  // onLinkClicked: (url: string) => void,
+  // shouldHandleLink?: (url: string) => boolean
```
What about removing this comments? What do you think?

dmnplb replied:
I wasn't sure if we could get rid of these methods, so I left them in. They were managed by the legacy Markdown component, but I'm not sure if they are still needed or not.

shadowsheep1 commented on ts/features/zendesk/screens/ZendeskSupportHelpCenter.tsx:
```diff
@@ -200,20 +196,20 @@ const ZendeskSupportHelpCenter = () => {
-  const [markdownContentLoaded, setMarkdownContentLoaded] = useState<boolean>(
-    !contextualHelpMarkdown
-  );
+  // const [markdownContentLoaded, setMarkdownContentLoaded] = useState<boolean>(
+  //   !contextualHelpMarkdown
+  // );
```
As above

shadowsheep1 commented on ts/features/zendesk/screens/ZendeskSupportHelpCenter.tsx:
```diff
@@ -200,20 +196,20 @@ const ZendeskSupportHelpCenter = () => {
-      return true;
-    }
+    contextualHelpMarkdown
+    // () => setMarkdownContentLoaded(true),
+    // constNull,
+    // _ => {
+    //   // when a link is clicked in the contextual help, terminate the workunit before the link will be handled (i.e: internal or external navigation)
+    //   workUnitComplete();
+    //   return true;
+    // }
```
As above

shadowsheep1 commented:
@dmnplb I did it a try and I can confirm that:
- Panic mode screen's still have the old UI
- The fallback FAQ mechanism is still working when (`content` -> `contextualData` -> is `None`)
- Using IOMarkdown is blazing fast 🤩. But I found a rendering regression for the IdP bullet list in "I forgot my SPID credential..." cc @ChrisMattew 
- It seems that in the first screen, in the Footer Action, the solid butto overflow the blank footer area. Am I right? <img src="https://github.com/user-attachments/assets/d5377b99-4720-4b0b-97bf-de78201ee9bb" width="150" target="_blank">


<details><summary>Video</summary>
<p>

| Test |
| - |
| <video src="https://github.com/user-attachments/assets/b6fb3437-08a5-4034-92a2-8c26875c5935" /> |

</p>
</details>

dmnplb commented:
@shadowsheep1 Thanks for the review. I have made some progress:

- Removed all the comments → https://github.com/pagopa/io-app/pull/5869/commits/d35e5886fbfcbd8a22881e2c427e7bdc3e99c0ab
- Removed all the legacy components from `ZendeskSeeReportsRouters` using the new `LoadingScreenContent` and `OperationResultScreenContent` → https://github.com/pagopa/io-app/pull/5869/commits/ca1a4a19c4a54bd69a20124c271ea027daf5ac84
> Panic mode screen's still have the old UI

Replaced the legacy component with `OperationResultScreenContent` → https://github.com/pagopa/io-app/pull/5869/commits/c56256d3486f304d8d087957479999e5190491b2

> It seems that in the first screen, in the actions footer, the solid button overflow the blank footer area. Am I right?

It's `FooterActions` if it's fixed and not managed by `IOScrollView`. I didn't use it at first because I needed a special header, but I realised that I could still use `useLayoutEffect` with `IOScrollView`. So I made the change in the following commit → https://github.com/pagopa/io-app/pull/5869/commits/385a460eba79e3354461832ba6b9f53c8e4b444a

Sadly, there are still open issues:
* Some tests are still failing
* I don't know if we always have to signal the end of the flow (trough `workUnitComplete`), even when the user taps an external link
* There's a type mismatch in the `IdpLoginScreen` component

shadowsheep1 reviewed (approved):
LGTM. 

For the markdown part, it's right how this new implementation render it, because what's wrong is the markdown itself coming from io-services-metadata (only the EN and DE part).
